File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/attachments/attachments.ts
import { When } from '@cucumber/fake-cucumber'
import fs from 'node:fs'

When('the string {string} is attached as {string}', async function (text: string, mediaType: string) {
  await this.attach(text, mediaType)
})

When('the string {string} is logged', async function (text: string) {
  await this.log(text)
})

When('text with ANSI escapes is logged', async function () {
  await this.log(
    'This displays a \x1b[31mr\x1b[0m\x1b[91ma\x1b[0m\x1b[33mi\x1b[0m\x1b[32mn\x1b[0m\x1b[34mb\x1b[0m\x1b[95mo\x1b[0m\x1b[35mw\x1b[0m'
  )
})

When('the following string is attached as {string}:', async function (mediaType: string, text: string) {
  await this.attach(text, mediaType)
})

When(
  'an array with {int} bytes is attached as {string}',
    async function (size: number, mediaType: string) {
    const data = [...Array(size).keys()]
    const buffer = Buffer.from(data)
    await this.attach(buffer, mediaType)
  }
)

When('a PDF document is attached and renamed', async function () {
  await this.attach(fs.createReadStream(__dirname + '/document.pdf'), {
      mediaType: 'application/pdf',
      fileName: 'renamed.pdf'
  })
})

When('a link to {string} is attached', async function (uri: string) {
    await this.link(uri)
})

File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/attachments/attachments.cs
using Reqnroll;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CucumberMessages.CompatibilityTests.CCK.attachments
{
    [Binding]
    internal class Attachments
    {
        private readonly IReqnrollOutputHelper reqnrollOutputHelper;

        internal Attachments(IReqnrollOutputHelper reqnrollOutputHelper)
        {
            this.reqnrollOutputHelper = reqnrollOutputHelper;
        }

        [When(@"the string {string} is attached as {string}")]
        public void WhenAttachTextAs(string text, string mediaType)
        {
            throw new PendingException("attaching strings with mediaTypes unsupported");
        }

        [When(@"the following string is attached as {string}:")]
        public void WhenAttachTextAsMultilineString(string mediaType, string text)
        {
            throw new PendingException("attaching strings with mediaTypes unsupported");
        }

        [When(@"an array with {int} bytes is attached as {string}")]
        public void WhenAttachByteArrayAs(int byteCount, string mediaType)
        {
            throw new PendingException("attaching byte arrays with mediaTypes unsupported");
        }

        [When(@"a PDF document is attached and renamed")]
        public void WhenAttachPdfDocumentRenamed()
        {
            throw new PendingException("attaching files and renaming them unsupported");
        }

        [When(@"a link to {string} is attached")]
        public void WhenAttachLinkTo(string url)
        {
            throw new PendingException("attaching links unsupported");
        }

        [When(@"the string {string} is logged")]
        public void WhenLogText(string text)
        {
            reqnrollOutputHelper.WriteLine(text);
        }

        [When(@"text with ANSI escapes is logged")]
        public void WhenTextWithANSIEscapedIsLogged()
        {
            reqnrollOutputHelper.WriteLine("This displays a \x1b[31mr\x1b[0m\x1b[91ma\x1b[0m\x1b[33mi\x1b[0m\x1b[32mn\x1b[0m\x1b[34mb\x1b[0m\x1b[95mo\x1b[0m\x1b[35mw\x1b[0m");
        }

    }
}
